bin/pace_collect.py

```diff
@@ -0,0 +1,161 @@
+        vasp_output_dict["ase_atoms"].append(ase_atoms)
+        vasp_output_dict["energy"].append(free_energy)
+        vasp_output_dict["forces"].append(forces)
+        
+        chemical_symbols = ase_atoms.get_chemical_symbols()
+        counter_elements = Counter(chemical_symbols)
+        reference_energy = 0.
+        for symbol, counter in counter_elements.items():
+            reference_energy += free_atom_energy_dict[symbol]*counter
+        
+        vasp_output_dict["energy_corrected"].append(free_energy - reference_energy)
+            
+    return vasp_output_dict
+
+
+class ParseKwargs(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, dict())
+        for value in values:
+            key, value = value.split(':')[0], float(value.split(':')[1]) 
+            getattr(namespace, self.dest)[key] = value
+
+
+def main(args):
+    
+    ##############################################################################################
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-wd", "--working-dir", help="top directory where keep calculations",
+                        type=str, default='.', dest="working_dir")
+        
+    parser.add_argument("--pickle-filename", help="pickle filename",
+                        type=str, default="collected.pckl.gzip", dest="pickle_filename")
+    
+    parser.add_argument('--free-atom-energy', help="dictionary of reference energies, Type: energy",
+                        nargs='*', dest="free_atom_energy", default=defaultdict(lambda:0), action=ParseKwargs)
+    
+    args_parse = parser.parse_args(args)
+    working_dir = os.path.abspath(args_parse.working_dir)
+    free_atom_energy_dict = args_parse.free_atom_energy
+    pickle_filename = args_parse.pickle_filename
+    
+    ##############################################################################################
+    logger.info('Following atomic reference energies will be subtracted : {}'.format(', '.join([str(k)+':'+str(v) for k,v in free_atom_energy_dict.items()])))
+   
+    vasprun_file = 'vasprun.xml'
+    outcar_file = 'OUTCAR'
+    data = {"name": [], "energy": [], "forces": [], "energy_corrected":[], "ase_atoms": []}
+
+    for root, _, filenames in walk_file_or_dir(working_dir):
+
+        if filenames and vasprun_file in filenames:
+            try:
+                vasp_output_dict = read_from_vasprun(root, vasprun_file, free_atom_energy_dict)
+                for key, value in vasp_output_dict.items():
+                    data[key] += value
+                logger.info('Data collected successfully from {} with entries {}'.format(os.path.join(root,vasprun_file),len(vasp_output_dict["name"])))
+            except Exception as e:
+                logger.error('Filename could not be read: {}'.format(str(e)))
+        elif filenames and not vasprun_file in filenames:
+            try:
```

Comment:
one try...except  block around vasrprun.xml or OUTCAR/CONTCAR if-else branches is better